wp-includes/template.php

```diff
@@ -171,3 +171,47 @@ function wp_service_worker_error_details_template( $output = '' ) {
 function wp_service_worker_error_message_placeholder() {
 	echo '<p><!--WP_SERVICE_WORKER_ERROR_MESSAGE--></p>';
 }
+
+/**
+ * Reload the offline page and check if user comes online.
+ *
+ * @since 0.7
+ */
+function wp_service_worker_offline_page_reload() {
+	if ( ! is_offline() ) {
+		return;
+	}
+	?>
+	<script>
+		/**
+		 * Listen to changes in the network state, reload when online.
+		 * This handles the case when the device is completely offline.
+		 */
+		window.addEventListener('online', () => {
+			window.location.reload();
+		});
+
+		/**
+		 * Check if the server is responding and reload the page if it is.
+		 * This handles the case when the device is online, but the server is offline or misbehaving.
+		 */
+		async function checkNetworkAndReload() {
+			try {
+				const response = await fetch('.', {method: 'HEAD'});
+				// Verify we get a valid response from the server
+				if (response.status >= 200 && response.status < 500) {
+					window.location.reload();
+					return;
+				}
+			} catch {
+				console.log('User is offline');
+			}
+			window.setTimeout(checkNetworkAndReload, 2500);
```

Comment:
I think we should also reset the counter after the count is quals to 5 or 6. There can be a possibility that the user or server comes back online and page reloading has to wait for another 1 or 2 minutes due to time delay. What do you think?

Reply:
@thelovekesh We can cap the number of retries, so after `n` number of retries, there won't be more.

Reply:
@westonruter Please let me know your thoughts on it.

Reply:
I think the exponential backoff will be enough. After 15 tries the timeout goes up to ~1 day, so it will effectively be disabled once the count goes high enough.